crypto/threads_none.c

```diff
@@ -17,6 +18,84 @@
 #  include <unistd.h>
 # endif
 
+struct rcu_lock_internal {
+    CRYPTO_RCU_LOCK lock; /* Must be first */
+    struct rcu_cb_item *cb_items;
+};
+
+CRYPTO_RCU_LOCK *ossl_rcu_lock_new(void)
+{
+    struct rcu_lock_internal *ilock;
+
+    ilock = OPENSSL_zalloc(sizeof(struct rcu_lock_internal));
+    return (CRYPTO_RCU_LOCK *)ilock;
+}
+
+void ossl_rcu_lock_free(CRYPTO_RCU_LOCK *lock)
+{
+    OPENSSL_free(lock);
+}
+
+void ossl_rcu_read_lock(CRYPTO_RCU_LOCK *lock)
+{
+    return;
+}
+
+void ossl_rcu_write_lock(CRYPTO_RCU_LOCK *lock)
+{
+    return;
+}
+
+void ossl_rcu_write_unlock(CRYPTO_RCU_LOCK *lock)
+{
+    return;
+}
+
+void ossl_rcu_read_unlock(CRYPTO_RCU_LOCK *lock)
+{
+    return;
+}
+
+void ossl_synchronize_rcu(CRYPTO_RCU_LOCK *lock)
+{
+    struct rcu_lock_internal *ilock = (struct rcu_lock_internal *)lock;
+    struct rcu_cb_item *items = ilock->cb_items;
+    struct rcu_cb_item *tmp;
+
+    ilock->cb_items = NULL;
+
+    while (items != NULL) {
+        tmp = items->next;
+        items->fn(items->data);
+        OPENSSL_free(items);
+        items = tmp;
+    }
+}
+
+void ossl_rcu_call(CRYPTO_RCU_LOCK *lock, rcu_cb_fn cb, void *data)
```

Comment:
we need it for deferred action in retirement.  The example case I added doesn't use it because changes are singular in nature, but if you want to batch a set of changes to a data structure, you use this to queue them behind a single synchrnoize_rcu.  See call_rcu in the kernel for a corollary.